token/js/test/e2e-2022/tokenMetadata.test.ts

```diff
@@ -0,0 +1,181 @@
+import { expect } from 'chai';
+
+import type { Connection, Signer } from '@solana/web3.js';
+import {
+    sendAndConfirmTransaction,
+    PublicKey,
+    Keypair,
+    SystemProgram,
+    Transaction,
+    TransactionInstruction,
+} from '@solana/web3.js';
+import { TOKEN_METADATA_DISCRIMINATOR, unpack } from '@solana/spl-token-metadata';
+
+import type { ExtensionType } from '../../src';
+import {
+    initializeTokenMetadata,
+    createInitializeMintInstruction,
+    getMint,
+    TOKEN_2022_PROGRAM_ID,
+    getExtensionData,
+    updateTokenMetadataField,
+    emitTokenMetadata,
+} from '../../src';
+import { TEST_PROGRAM_ID, newAccountWithLamports, getConnection } from '../common';
+
+const TEST_TOKEN_DECIMALS = 2;
+
+// TODO: Remove or move to src. Temp function so we can repack to use spl-token-metadata
+const lengthBuffer = (buffer: Buffer | Uint8Array): Buffer => {
+    const length = Buffer.alloc(4);
+    length.writeUIntLE(buffer.length, 0, 4);
+    return length;
+};
+
+describe('Token Metadata', () => {
```

Comment:
I had difficulty with the unpacking of the metadata. I got it to work, but don't understand why and thinking perhaps i got lucky?

I assumed its needs to be done, but please correct me if i'm wrong.